perception/radar_crossing_objects_noise_filter/src/radar_crossing_objects_noise_filter_node/radar_crossing_objects_noise_filter_node.cpp

```diff
@@ -0,0 +1,148 @@
+#endif
+
+namespace
+{
+template <class T>
+bool update_param(
+  const std::vector<rclcpp::Parameter> & params, const std::string & name, T & value)
+{
+  const auto itr = std::find_if(
+    params.cbegin(), params.cend(),
+    [&name](const rclcpp::Parameter & p) { return p.get_name() == name; });
+
+  // Not found
+  if (itr == params.cend()) {
+    return false;
+  }
+
+  value = itr->template get_value<T>();
+  return true;
+}
+}  // namespace
+
+namespace radar_crossing_objects_noise_filter
+{
+using autoware_auto_perception_msgs::msg::DetectedObject;
+using autoware_auto_perception_msgs::msg::DetectedObjects;
+
+RadarCrossingObjectsNoiseFilterNode::RadarCrossingObjectsNoiseFilterNode(
+  const rclcpp::NodeOptions & node_options)
+: Node("radar_crossing_objects_noise_filter", node_options)
+{
+  // Parameter Server
+  set_param_res_ = this->add_on_set_parameters_callback(
+    std::bind(&RadarCrossingObjectsNoiseFilterNode::onSetParam, this, std::placeholders::_1));
+
+  // Node Parameter
+  node_param_.angle_threshold = declare_parameter<double>("angle_threshold", 1.0472);
+  node_param_.velocity_threshold = declare_parameter<double>("velocity_threshold", 3.0);
+
+  // Subscriber
+  sub_objects_ = create_subscription<DetectedObjects>(
+    "~/input/objects", rclcpp::QoS{1},
+    std::bind(&RadarCrossingObjectsNoiseFilterNode::onObjects, this, std::placeholders::_1));
+
+  // Publisher
+  pub_filtered_objects_ = create_publisher<DetectedObjects>("~/output/filtered_objects", 1);
+  pub_noise_objects_ = create_publisher<DetectedObjects>("~/output/noise_objects", 1);
+}
+
+void RadarCrossingObjectsNoiseFilterNode::onObjects(
+  const DetectedObjects::ConstSharedPtr objects_data_)
+{
+  DetectedObjects filtered_objects;
+  DetectedObjects noise_objects;
+  filtered_objects.header = objects_data_->header;
+  noise_objects.header = objects_data_->header;
+
+  for (const auto & object : objects_data_->objects) {
+    if (isNoise(object)) {
+      noise_objects.objects.emplace_back(object);
+    } else {
+      filtered_objects.objects.emplace_back(object);
+    }
+  }
+  // publish
+  pub_filtered_objects_->publish(filtered_objects);
+  pub_noise_objects_->publish(noise_objects);
+}
+
+rcl_interfaces::msg::SetParametersResult RadarCrossingObjectsNoiseFilterNode::onSetParam(
+  const std::vector<rclcpp::Parameter> & params)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+
+  try {
+    // Node Parameter
+    {
+      auto & p = node_param_;
+
+      // Update params
+      update_param(params, "angle_threshold", p.angle_threshold);
+      update_param(params, "velocity_threshold", p.velocity_threshold);
+    }
+  } catch (const rclcpp::exceptions::InvalidParameterTypeException & e) {
+    result.successful = false;
+    result.reason = e.what();
+    return result;
+  }
+
+  result.successful = true;
+  result.reason = "success";
+  return result;
+}
+
+bool RadarCrossingObjectsNoiseFilterNode::isNoise(const DetectedObject & object)
```

Comment:
This process highly depends on assumption that `object` is in the `base_link` frame.
Add comment or use `perception_utils::transformObjects` to fix frame to `base_link` will more friendly for the code readers.